test/Utilities.test.js

```diff
@@ -117,6 +117,15 @@ contract("Utilities", function (accounts) {
 
         });
 
+        // This is a bug in the Utilities.sol uintToHexStr function
+        it('it should return the address without leading zero [BUG]', async function () {
```

Comment:
I believe there is a bug in this solidity library.

Reply:
So solidity removes leading 0s for uints because it think it assumes that is the sign. 
```
uint payload = 0x04048d5cc923ce97f9265dcf70f2fea47319c6;
uint payload2 = 0x4048d5cc923ce97f9265dcf70f2fea47319c6;
require(payload == payload2);
```
this will pass

Reply:
Yes for `uint` comparisons, it will fail in places where we convert to hexstring and compare. In particular the one that I didnt fix is the tokenURI comparison, where there is a serial concatenation of strings and is built within the smart contract.

Reply:
```

Reply:
we need to do a address to string method. For uints It works correctly IMO and we should not have uints with leading 0s